internal/pkg/vault/management.go

```diff
@@ -0,0 +1,193 @@
+	}
+
+	c.lc.Infof("vault health check HTTP status: StatusCode: %d", code)
+	return code, nil
+}
+
+func (c *Client) Init(secretThreshold int, secretShares int) (types.InitResponse, error) {
+	c.lc.Infof("vault init strategy (SSS parameters): shares=%d threshold=%d",
+		secretShares,
+		secretThreshold)
+
+	request := InitRequest{
+		SecretShares:    secretShares,
+		SecretThreshold: secretThreshold,
+	}
+
+	response := types.InitResponse{}
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            "",
+		Method:               http.MethodPost,
+		Path:                 InitAPI,
+		JSONObject:           &request,
+		BodyReader:           nil,
+		OperationDescription: "initialize secret store",
+		ExpectedStatusCode:   http.StatusOK,
+		ResponseObject:       &response,
+	})
+
+	return response, err
+}
+
+func (c *Client) Unseal(keys []string, keysBase64 []string) error {
+	c.lc.Infof("Vault unsealing Process. Applying key shares.")
+
+	secretShares := len(keys)
+
+	keyCounter := 1
+	for _, key := range keysBase64 {
+		request := UnsealRequest{Key: key}
+		response := UnsealResponse{}
+
+		_, err := c.doRequest(RequestArgs{
+			AuthToken:            "",
+			Method:               http.MethodPost,
+			Path:                 UnsealAPI,
+			JSONObject:           &request,
+			BodyReader:           nil,
+			OperationDescription: "unseal secret store",
+			ExpectedStatusCode:   http.StatusOK,
+			ResponseObject:       &response,
+		})
+
+		if err != nil {
+			c.lc.Error(fmt.Sprintf("Error applying key share %d/%d: %s", keyCounter, secretShares, err.Error()))
+			return err
+		}
+
+		c.lc.Info(fmt.Sprintf("Vault key share %d/%d successfully applied.", keyCounter, secretShares))
+		if !response.Sealed {
+			c.lc.Info("Vault key share threshold reached. Unsealing complete.")
+			return nil
+		}
+		keyCounter++
+	}
+
+	return fmt.Errorf("%d", 1)
+}
+
+func (c *Client) InstallPolicy(token string, policyName string, policyDocument string) error {
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPut,
+		Path:                 fmt.Sprintf(CreatePolicyPath, url.PathEscape(policyName)),
+		JSONObject:           UpdateACLPolicyRequest{Policy: policyDocument},
+		BodyReader:           nil,
+		OperationDescription: "install policy",
+		ExpectedStatusCode:   http.StatusNoContent,
+		ResponseObject:       nil,
+	})
+
+	return err
+}
+
+func (c *Client) EnableKVSecretEngine(token string, mountPoint string, kvVersion string) error {
+	urlPath := path.Join(MountsAPI, mountPoint)
+	parameters := EnableSecretsEngineRequest{
+		Type:        KeyValue,
+		Description: "key/value secret storage",
+		Options: &SecretsEngineOptions{
+			Version: kvVersion,
+		},
+	}
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPost,
+		Path:                 urlPath,
+		JSONObject:           parameters,
+		BodyReader:           nil,
+		OperationDescription: "update mounts",
```

Comment:
```suggestion
		OperationDescription: "update mounts for KV",
```

Reply:
Humm, I had that... no sure where it went.